tests/test_context_collection.py

```diff
@@ -0,0 +1,376 @@
+# SPDX-FileCopyrightText: 2022 Copyright DB Netz AG and the capellambse-context-diagrams contributors
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import typing as t
+
+import capellambse
+import pytest
+import yaml
+
+from capellambse_context_diagrams import collectors
+from capellambse_context_diagrams.collectors import default, portless
+
+# pylint: disable-next=relative-beyond-top-level
+from .conftest import TEST_ROOT
+
+CONTEXT_PATH = TEST_ROOT / "context"
+
+ExpectedConnectionInfo = dict[t.Literal["input", "output"], list[t.Any]]
+
+
+def load_context_description(name: str) -> dict[str, t.Any]:
+    return yaml.safe_load((CONTEXT_PATH / f"{name}.yaml").read_bytes())
+
+
+def build_expected_context(
+    model: capellambse.MelodyModel, builder: ContextInfoBuilder
+) -> list[collectors.ContextInfo]:
+    if builder.collector == portless.context_collector:
```

Comment:
Why this condition here, instead of just making two functions and calling the correct one?

I think overall it would even be better (as in, more readable and easier to understand) to make a function (or two) returning a `pytest.param` which directly contains `expected_context` and the two extra arguments to `collect_free_context`.